### PR TITLE
fix: set OP enveloped tx field

### LINKED ISSUE #7648: anvil crashes with `thread 'tokio-runtime-worker' panic`
### Component

Anvil

### Have you ensured that all of these are up to date?

- [X] Foundry
- [X] Foundryup

### What version of Foundry are you on?

forge 0.2.0 (bdc04c2 2024-04-12T00:21:19.564747000Z)

### What command(s) is the bug in?

anvil

### Operating System

macOS (Apple Silicon)

### Describe the bug

I'm working on an integration test which involves deploying [`nana-core`](https://github.com/Bananapus/nana-core) on local anvil forks of mainnet and optimism. We're using [sphinx](https://github.com/sphinx-labs/sphinx) to manage deploys, which works well on the mainnet fork. On Optimism, however, this fails. To reproduce:

1. Start an anvil server with

```bash 
anvil -f "<OPTIMISM_RPC_HERE>" --optimism
```

2. Start the deployment with

```bash
RPC_OPTIMISM_MAINNET="http://localhost:8545" npx sphinx deploy script/Deploy.s.sol --verify false --network optimism
```

This runs the foundry deploy script [here](https://github.com/Bananapus/nana-core/blob/main/script/Deploy.s.sol).

sphinx is able to prepare the transacrions, but upon confirmation (during `Deploying Gnosis Safe and Sphinx Module...`), anvil crashes with the following output:

```
thread 'tokio-runtime-worker' panicked at crates/anvil/src/eth/backend/executor.rs:298:29:
Failed to execute transaction. This is a bug.
 Custom("[OPTIMISM] Failed to load enveloped transaction.")
stack backtrace:
   0: _rust_begin_unwind
   1: core::panicking::panic_fmt
   2: <&mut anvil::eth::backend::executor::TransactionExecutor<DB,Validator> as core::iter::traits::iterator::Iterator>::next
   3: anvil::eth::backend::mem::Backend::do_mine_block::{{closure}}
   4: tokio::runtime::task::raw::poll
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
```

I've attached the output with `RUST_BACKTRACE` set to full:

[report.txt](https://github.com/foundry-rs/foundry/files/14961602/report.txt)

I'll also let the sphinx team know about this. They may have some ideas.

### PR DESCRIPTION
**Pull request by mattsse**
should close #7648

### CODE REVIEW

File: crates/anvil/src/eth/backend/executor.rs [hunk 1]
```diff
@@ -118,7 +118,7 @@ impl<'a, DB: Db + ?Sized, Validator: TransactionValidator> TransactionExecutor<'
         let difficulty = self.block_env.difficulty;
         let beneficiary = self.block_env.coinbase;
         let timestamp = self.block_env.timestamp.to::<u64>();
-        let base_fee = if (self.cfg_env.handler_cfg.spec_id as u8) >= (SpecId::LONDON as u8) {
+        let base_fee = if self.cfg_env.handler_cfg.spec_id >= SpecId::LONDON {
```
Comment (DaniPopes) [inline crates/anvil/src/eth/backend/executor.rs:121]: spec_id.is_enabled_in(LONDON)

File: crates/anvil/src/eth/backend/executor.rs [hunk 2]
```diff
@@ -208,6 +208,12 @@ impl<'a, DB: Db + ?Sized, Validator: TransactionValidator> TransactionExecutor<'
     }
 
     fn env_for(&self, tx: &PendingTransaction) -> EnvWithHandlerCfg {
+        let mut tx_env = tx.to_revm_tx_env();
```
Comment (klkvr) [inline crates/anvil/src/eth/backend/executor.rs:211]: we are still passing `tx.to_revm_tx_env()` into `new_with_cfg_env` instead of `tx_env` var

**Review by klkvr — COMMENTED** [review]
tested locally and it seems that we also occur those errors in `eth_call` and probably `eth_estimateGas`. not sure how to handle this as we can't obtain enveloped tx from a tx request

File: crates/anvil/src/eth/backend/executor.rs (hunk 2 above)
Comment (mattsse) [reply crates/anvil/src/eth/backend/executor.rs:211]: gotcha, mind taking over this?

Comment (klkvr) [reply crates/anvil/src/eth/backend/executor.rs:211]: yep sure

**Comment by klkvr** [comment]
@mattsse repro from #7648 passes now

**Pull request merged**